### PR TITLE
Add Radius mode to Identify plugin

### PR DESCRIPTION
**Pull request by benoitblanc**
Hi,

This PR aims to add mode `Radius` to Identify plugin. User can identify features around a location in the map regarding a radius. A marker and a circle are added to the map when identifying with this mode.

image

image

To enable it, we have to add the new mode in TopBar configuration in `static/config.json`:

```json
      {
        "name": "TopBar",
        "cfg": {
          "menuItems": [
            {"key": "ThemeSwitcher", "icon": "themes", "shortcut": "alt+shift+t"},
            {"key": "LayerTree", "icon": "layers", "shortcut": "alt+shift+l"},
            {"key": "LayerCatalog", "icon": "catalog", "shortcut": "alt+shift+c"},
            {"key": "Share", "icon": "share", "shortcut": "alt+shift+s"},
            {"key": "Bookmark", "icon": "bookmark", "shortcut": "alt+shift+b"},
            {"key": "Tools", "icon": "tools", "subitems": [
              {"key": "Identify", "icon": "identify_region", "mode": "Region"},
              {"key": "Identify", "icon": "identify_radius", "mode": "Radius"}, <=====================
              {"key": "TimeManager", "icon": "clock"},
              {"key": "Measure", "icon": "measure"},
              {"key": "Redlining", "icon": "draw"},
              {"key": "Editing", "icon": "editing"},
              {"key": "FeatureForm", "icon": "featureform"},
              {"key": "AttributeTable", "icon": "editing"},
              {"key": "DxfExport", "icon": "dxfexport"},
              {"key": "RasterExport", "icon": "rasterexport"}
            ]},
            {"key": "Print", "icon": "print", "shortcut": "alt+shift+p"},
            {"key": "Help", "icon": "info", "shortcut": "alt+shift+h"},
            {"key": "Settings", "icon": "cog"},
            {"key": "ExternalLink", "icon": "link", "url": "http://example.com?extent=$e$", "target": "iframe"}
          ],
          "toolbarItems": [
            {"key": "Measure", "mode": "LineString", "icon": "measure_line"},
            {"key": "Measure", "mode": "Polygon", "icon": "measure_polygon"},
            {"key": "Print", "icon": "print"},
            {"key": "Identify", "icon": "identify_region", "mode": "Region"},
            {"key": "Identify", "icon": "identify_radius", "mode": "Radius"}, <==================
            {"key": "FeatureForm", "icon": "featureform"},
            {"key": "TimeManager", "icon": "clock"}
          ],
          "searchOptions": {
            "minScaleDenom": 1000,
            "showProviderSelection": true,
            "providerSelectionAllowAll": true,
            "zoomToLayers": false,
            "showProvidersInPlaceholder": false
          },
          "appMenuClearsTask": true,
          "appMenuFilterField": true,
          "appMenuVisibleOnStartup": false,
          "logoUrl": "/",
          "appMenuShortcut": "alt+shift+m",
          "toolbarItemsShortcutPrefix": "alt+shift"
        }
      },
```

Thanks

### CODE REVIEW

File: plugins/Identify.jsx [hunk 1]
```diff
@@ -61,7 +66,18 @@ class Identify extends React.Component {
...
+        radius: 50,
+        units: 'meters'
+    }
+    constructor(props, context) {
+        super(props, context);
+        this.state.layer = {
+            id: "buffer",
+            title: LocaleUtils.tr("identify.bufferlayername"),
+            role: LayerRole.USERLAYER
+        };
```
Comment (manisandro) [inline plugins/Identify.jsx:80]: `buffer` is a bit too generic as ID, owuld call it `identifyradiusbuffer` or similar.

There is probably no point having this layer show in the layertree, so I'd set `role: LayerRole.SELECTION` (which are automatically omitted from the LayerTree) and omit the `title`.

I wouldn't store this in the component state, rather just as a local variable in `computeBuffer()`.

**Comment by manisandro** [comment]
Nice work!

**Comment by benoitblanc** [comment]
Thanks @manisandro for the review ! I have pushed new commits to resolve your remarks.

File: plugins/Identify.jsx [hunk 2]
```diff
@@ -61,7 +66,9 @@ class Identify extends React.Component {
     state = {
         mode: 'Point',
         identifyResults: null,
-        pendingRequests: 0
+        pendingRequests: 0,
+        radius: 50,
+        units: 'meters'
```
Comment (manisandro) [inline plugins/Identify.jsx:71]: Small nitpick: units -> radiusUnits, to give implicit context to the state variable.

**Comment by manisandro** [comment]
One last nitpick, rest looks good!

**Comment by manisandro** [comment]
Thanks!

**Pull request merged**